packages/agents/lighthouse/src/tasks/propose/operations/proposeHub.ts

```diff
@@ -86,10 +88,14 @@ export const proposeHub = async () => {
         if (!snapshotRoot) {
           throw new NoSnapshotRoot(domain, requestContext, methodContext);
         }
-        const latestDbSnapshotId = Math.floor(snapshotRoot.timestamp / config.snapshotDuration);
-        if (latestSnapshotId > latestDbSnapshotId) {
-          latestSnapshotRoot = await getCurrentOutboundRoot(domain, requestContext);
-
+        const latestDbSnapshotId = +snapshotRoot.id;
+        latestSnapshotRoot = await getCurrentOutboundRoot(domain, requestContext);
+        const outboundRootTimestamp = await database.getOutboundRootTimestamp(domain, latestSnapshotRoot);
+        if (
+          latestSnapshotId > latestDbSnapshotId &&
+          outboundRootTimestamp &&
+          outboundRootTimestamp < latestSnapshotTimestamp
+        ) {
```

Comment:
we need to have some comments here otherwise we will likely not remember the exact reasoning

Reply:
Indeed, added comments explaining the boundary case.